processor/groupbyattrsprocessor/processor_test.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package groupbyattrsprocessor
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+)
+
+var (
+	logger, _ = zap.NewDevelopment()
+	attrMap   = prepareAttributeMap()
+)
+
+func prepareAttributeMap() pdata.AttributeMap {
+	attributeValues := map[string]pdata.AttributeValue{
+		"xx": pdata.NewAttributeValueString("aa"),
+		"yy": pdata.NewAttributeValueInt(11),
+	}
+
+	attrMap := pdata.NewAttributeMap()
+	attrMap.InitFromMap(attributeValues)
+
+	attrMap.Sort()
+	return attrMap
+}
+
+func prepareResource(attrMap pdata.AttributeMap, selectedKeys []string) pdata.Resource {
+	res := pdata.NewResource()
+	for _, key := range selectedKeys {
+		val, found := attrMap.Get(key)
+		if found {
+			res.Attributes().Insert(key, val)
+		}
+	}
+	res.Attributes().Sort()
+	return res
+}
+
+func filterAttributeMap(attrMap pdata.AttributeMap, selectedKeys []string) pdata.AttributeMap {
+	filteredAttrMap := pdata.NewAttributeMap()
+	for _, key := range selectedKeys {
+		val, _ := attrMap.Get(key)
+		filteredAttrMap.Insert(key, val)
+	}
+	filteredAttrMap.Sort()
+	return filteredAttrMap
+}
+
+func TestLogAttributesExtracted(t *testing.T) {
```

Comment:
Hmm IMHO either change the test name or make separate tests for logs and spans.

For now the test name mentions logs but tests itself performs checks on both the spans and logs.